Gifski/Gifski.swift

```diff
@@ -44,6 +44,8 @@ final class Gifski {
 		}
 	}
 
+	private static var gifData: NSMutableData?
```

Comment:
We are holding strongly here so we can have an influence over the lifetime of gifData.

Reply:
This is interesting! I thought that because we are using the variable in the closure we add to the retain count and so it should do the same thing as this trick does?

Also I wonder if we should just treat the `Gifski` as a proper object with a lifecycle, without using global methods/properties now. Initializing an object in a `ConversionViewController` instead. What do you guys think?

Reply:
@sunshinejr I agree. The `Gifski` class only exist now as I wanted a namespace for the `run` method. Now that we have some state, it makes more sense to make an instance of it. We might be able to reduce the size of the `run` method this way too.

Reply:
Ye definitely done is separate PR. Will add some in line documentation now.

Reply:
> This is interesting! I thought that because we are using the variable in the closure we add to the retain count and so it should do the same thing as this trick does?
> 
> Also I wonder if we should just treat the `Gifski` as a proper object with a lifecycle, without using global methods/properties now. Initializing an object in a `ConversionViewController` instead. What do you guys think?

Ye, I'm used to swift (and Obj-C) closure's / blocks capturing stuff. 

Interestingly, the docs for Gifski library suggests the `setWriteCallback` block may take a short while to finish off the last task it was doing. So it seems to be doing a bit of work whilst the surrounding lexical scope is disappearing.  So holding on to gifData outside of that lexical scope solves that problem, for now.